hippo4j-core/src/main/java/cn/hippo4j/core/plugin/impl/MaximumActiveThreadCountChecker.java

```diff
@@ -0,0 +1,174 @@
+ *
+ * @author huangchengxing
+ * @see IllegalMaximumActiveCountException
+ */
+@Slf4j
+public class MaximumActiveThreadCountChecker implements ExecuteAwarePlugin {
+
+    /**
+     * Thread pool.
+     */
+    public final ThreadPoolPluginSupport threadPoolPluginSupport;
+
+    /**
+     * Whether to re-deliver the task to the thread pool after the maximum number of threads is exceeded.
+     */
+    private final AtomicBoolean enableSubmitTaskAfterCheckFail;
+
+    /**
+     * Create {@link MaximumActiveThreadCountChecker}.
+     *
+     * @param threadPoolPluginSupport thread pool
+     * @param enableSubmitTaskAfterCheckFail whether to re-deliver the task to the thread pool after the maximum number of threads is exceeded.
+     */
+    public MaximumActiveThreadCountChecker(ThreadPoolPluginSupport threadPoolPluginSupport, boolean enableSubmitTaskAfterCheckFail) {
+        this.threadPoolPluginSupport = threadPoolPluginSupport;
+        this.enableSubmitTaskAfterCheckFail = new AtomicBoolean(enableSubmitTaskAfterCheckFail);
+    }
+
+    /**
+     * Set whether to re-deliver the task to the thread pool after the maximum number of threads is exceeded.
+     *
+     * @param enableSubmitTaskAfterCheckFail whether to re-deliver the task to the thread pool after the maximum number of threads is exceeded.
+     */
+    public void setEnableSubmitTaskAfterCheckFail(boolean enableSubmitTaskAfterCheckFail) {
+        this.enableSubmitTaskAfterCheckFail.set(enableSubmitTaskAfterCheckFail);
+    }
+
+    /**
+     * Get whether to re-deliver the task to the thread pool after the maximum number of threads is exceeded.
+     *
+     * @return whether to re-deliver the task to the thread pool after the maximum number of threads is exceeded.
+     */
+    public boolean isEnableSubmitTaskAfterCheckFail() {
+        return enableSubmitTaskAfterCheckFail.get();
+    }
+
+    /**
+     * Get plugin runtime info.
+     *
+     * @return plugin runtime info
+     */
+    @Override
+    public PluginRuntime getPluginRuntime() {
+        return new PluginRuntime(getId())
+                .addInfo("enableSubmitTaskAfterCheckFail", enableSubmitTaskAfterCheckFail.get());
+    }
+
+    /**
+     * <p>Check the maximum number of threads in the thread pool before the task is executed,
+     * if the maximum number of threads is exceeded,
+     * an {@link IllegalMaximumActiveCountException} will be thrown.
+     *
+     * @param thread   thread of executing task
+     * @param runnable task
+     * @throws IllegalMaximumActiveCountException thread if the maximum number of threads is exceeded
+     * @see ThreadPoolExecutor#beforeExecute
+     */
+    @Override
+    public void beforeExecute(Thread thread, Runnable runnable) {
+        ThreadPoolExecutor threadPoolExecutor = threadPoolPluginSupport.getThreadPoolExecutor();
+        int activeCount = threadPoolExecutor.getActiveCount();
```

Comment:
The `getActiveCount` method is locked, is there a better performance implementation?

Reply:
Is it possible to execute by timer?